File: src/bot_general.js

const { Client } = require('discord.js');
const { playAudio } = require('./commands/audio');
const { existsSync, readdirSync } = require('fs');
const { basename } = require('path')
const { searchDrink, searchStore } = require('./commands/drinks.js');

const client = new Client();
const prefix = '!';
const pathToAudios = "../sounds";
const audios = existsSync(pathToAudios) ? 
    readdirSync(pathToAudios)
    .filter(fileName => fileName.includes('.mp3')) 
    .map(fileName => basename(fileName, '.mp3')) : null;

client.login(process.env.GENERAL_BOT);
client.on('ready', () => {
    console.log("General Bot successfully connected.");
});


client.on('message', (msg) => {
    if (msg.author.bot) return;
    if (!msg.content.startsWith(prefix)) return;

    const args = msg.content.slice(prefix.length).trim().split(/ +/g);
    const command = args.shift().toLowerCase();

    switch(command) {
        case 'search': 
            searchDrink(args, msg.channel);
            break;
        
        case 'locate': 
            searchStore(args, msg.channel);
            break;
        
        default:
            const audio = audios.find(audio => audio === command);
            if (audio) 
                playAudio(msg, `${pathToAudios}/${audio}.mp3`);
            break;
    }

    
})

File: python/bot_general.py
import os

from dotenv import load_dotenv
from discord.ext import commands


COMMAND_PREFIX = '!'

bot = commands.Bot(command_prefix=COMMAND_PREFIX)

for filename in os.listdir('./cogs'):
    if filename.endswith('.py'):
        print(filename)
        bot.load_extension(f"cogs.{filename[:-3]}")

    print("General bot has successfully connected!")
bot.run('ODAyMjY3NDY1Nzk4OTA5OTYy.YAsvzQ.VAvYUg5fparnQCzH-dlgjv11vmA')


